torch/testing/_internal/opinfo/definitions/signal.py

```diff
@@ -0,0 +1,269 @@
+import random
+import unittest
+
+from itertools import product
+from typing import List
+
+import torch
+from torch.testing._internal.common_dtype import all_types_and
+from torch.testing._internal.common_utils import TEST_SCIPY
+from torch.testing._internal.opinfo.core import (
+    DecorateInfo,
+    ErrorInput,
+    OpInfo,
+    SampleInput,
+)
+
+if TEST_SCIPY:
+    import scipy.signal
+
+
+def sample_inputs_window(op_info, device, dtype, requires_grad, *, kws=({},), **kwargs):
+    _kwargs = dict(
+        kwargs, **{"device": device, "dtype": dtype, "requires_grad": requires_grad}
+    )
+    sizes = [0, 1, 2, 5, 10, 50, 100, 1024, 2048]
+    for size, periodic, k in product(sizes, (True, False), kws):
+        yield SampleInput(size, periodic=periodic, **k, **_kwargs)
+
+
+def sample_inputs_gaussian_window(op_info, device, dtype, requires_grad, **kwargs):
+    kws = [{"std": random.uniform(0, 3)} for _ in range(50)]
+    yield from sample_inputs_window(
+        op_info, device, dtype, requires_grad, kws=kws, **kwargs
+    )
+
+
+def sample_inputs_exponential_window(op_info, device, dtype, requires_grad, **kwargs):
+    kws = [{"center": None, "tau": random.uniform(0, 10)} for _ in range(50)]
+    yield from sample_inputs_window(
+        op_info, device, dtype, requires_grad, kws=kws, **kwargs
+    )
+
+
+def error_inputs_window(op_info, device, **kwargs):
+    tmp_kwargs = dict(
+        kwargs,
+        **{
+            "device": device,
+        },
+    )
+
+    yield ErrorInput(
+        SampleInput(-1, kwargs=tmp_kwargs),
+        error_type=ValueError,
+        error_regex="requires non-negative window_length, got window_length=-1",
+    )
+
+    tmp_kwargs = dict(
+        kwargs,
+        **{
+            "layout": torch.sparse_coo,
+        },
+    )
+
+    yield ErrorInput(
+        SampleInput(3, kwargs=tmp_kwargs),
+        error_type=ValueError,
+        error_regex="is not implemented for sparse types, got: torch.sparse_coo",
+    )
+
+    tmp_kwargs = kwargs
+    tmp_kwargs.update(
+        {
+            "dtype": torch.long,
+        }
+    )
+
+    yield ErrorInput(
+        SampleInput(3, kwargs=tmp_kwargs),
+        error_type=ValueError,
+        error_regex="expects floating point dtypes, got: torch.int64",
+    )
+
+
+def error_inputs_exponential_window(op_info, device, **kwargs):
+    tmp_kwargs = dict(kwargs, **{"dtype": torch.float32})
```

Comment:
I don't think doing this kwarg treatment is necessary really. simply generating the relevant samples that would lead to the errors and not handling the `**kwargs` would be just fine. The `**kwargs` are just here to provide an extensibility point for some tests to pass some extra parameters for some particular OpInfos. You shouldn't care about them thouhg.

For the record, a better way of writing this line would be
```python
    tmp_kwargs = dict(kwargs, dtype=torch.float32)
```

Reply:
I resolved this.